### PR TITLE
Step Modal: Select a Template

### PR DESCRIPTION
**Pull request by airensurzyn**
Add functionality for TextEditor to: 
- Select Template from Templates List
- Save as Template

### CODE REVIEW

File: client/src/Components/Dialog/DataSelectDialog.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import {
+  List,
+  ListItem,
+  ListItemText,
+  DialogTitle,
+  Dialog,
+  Divider
+} from "@material-ui/core";
+
+const DataSelectDialog = props => {
```
Comment (skassam21) [inline client/src/Components/Dialog/DataSelectDialog.js:11]: This component is what I was looking for! Good job, this is essentially a presentational component

File: client/src/Views/Home/Steps/AddStep.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    overflow: "hidden",
+    height: "100%",
+    padding: "1rem"
+  },
+  emailContainer: {
+    margin: "1rem 2rem"
+  }
+}));
+
+const Step = props => {
```
Comment (Disquietude) [inline client/src/Views/Home/Steps/AddStep.js]: This component currently shouldn't exist. This looks like @umbauk 's previous implementation of the text editor, which I renamed to `StepEditor` and moved to `/Views/Home/Campaigns/Campaign`

File: client/src/Views/Home/Campaigns/Campaign/CampaignHeader.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const CampaignHeader = props => {
+  const classes = useStyles();
+
+  const handleAddProspects = () => {
+    props.history.push(`/home/prospects`);
+  };
+
+  const handleAddStep = () => {
+    props.history.push(`${props.match.url}/step`);
+  };
```
Comment (Disquietude) [inline client/src/Views/Home/Campaigns/Campaign/CampaignHeader.js]: Is this component being used anywhere? Currently there are only two pages (Summary and Prospects) for the individual campaign detail and both of them already have header components. Also, opening the editor should no longer be done by changing the browser location directly.

**Review by Disquietude — COMMENTED** [review]
Looks good overall. Some of the components look unnecessary or unused, however.

File: client/src/Views/Home/Steps/AddStep.js (hunk 2 above)
Comment (airensurzyn) [reply client/src/Views/Home/Steps/AddStep.js]: Hmmm, maybe I didn't merge dev before I pushed. I'll check it out

File: client/src/Views/Home/Campaigns/Campaign/CampaignHeader.js (hunk 3 above)
Comment (airensurzyn) [reply client/src/Views/Home/Campaigns/Campaign/CampaignHeader.js]: Yes, also probably due to not merging dev correctly.

**Pull request merged**